packages/vscode-extension/src/debugging/ProxyDebugAdapter.ts

```diff
@@ -0,0 +1,299 @@
+  ): void {
+    response.body = response.body || {};
+
+    response.body.supportsConfigurationDoneRequest = true;
+    response.body.supportsEvaluateForHovers = true;
+    response.body.supportTerminateDebuggee = false;
+    response.body.supportsCancelRequest = true;
+
+    response.body.exceptionBreakpointFilters = [
+      {
+        filter: "all",
+        label: "Caught Exceptions",
+        default: false,
+        supportsCondition: true,
+        description: "Breaks on all throw errors, even if they're caught later.",
+        conditionDescription: 'error.name == "MyError"',
+      },
+      {
+        filter: "uncaught",
+        label: "Uncaught Exceptions",
+        default: false,
+        supportsCondition: true,
+        description: "Breaks only on errors or promise rejections that are not handled.",
+        conditionDescription: 'error.name == "MyError"',
+      },
+    ];
+
+    this.sendResponse(response);
+  }
+
+  protected async attachRequest(
+    response: DebugProtocol.AttachResponse,
+    args: DebugProtocol.AttachRequestArguments & {
+      sourceMapPathOverrides: Record<string, string>;
+      websocketAddress: string;
+    },
+    request?: DebugProtocol.Request
+  ) {
+    await this.cdpProxy.initializeServer();
+
+    let didStartSessionHandler: Disposable | null = vscode.debug.onDidStartDebugSession(
+      (session) => {
+        if (session.type === CHILD_SESSION_TYPE) {
+          this.nodeDebugSession = session;
+          didStartSessionHandler?.dispose();
+          didStartSessionHandler = null;
+        }
+      }
+    );
+
+    try {
+      const childSessionStarted = await debug.startDebugging(
+        undefined,
+        {
+          type: CHILD_SESSION_TYPE,
+          name: "Radon IDE Debugger",
+          request: "attach",
+          port: this.cdpProxy.port,
+          continueOnAttach: true,
+          sourceMapPathOverrides: args.sourceMapPathOverrides,
+          resolveSourceMapLocations: ["**", "!**/node_modules/!(expo)/**"],
+          skipFiles: [
+            "**/extension/lib/**/*.js",
+            "**/vscode-extension/lib/**/*.js",
+            "**/ReactFabric-dev.js",
+            "**/ReactNativeRenderer-dev.js",
+            "**/node_modules/**/*",
+            "!**/node_modules/expo-router/**/*",
+          ],
+        },
+        {
+          suppressDebugStatusbar: true,
+          suppressDebugView: true,
+          suppressDebugToolbar: true,
+          suppressSaveBeforeStart: true,
+          parentSession: this.session,
+          consoleMode: vscode.DebugConsoleMode.MergeWithParent,
+          lifecycleManagedByParent: true,
+          compact: true,
+        }
+      );
+
+      if (!childSessionStarted) {
+        this.sendErrorResponse(
+          response,
+          { format: "Failed to attach debugger session", id: 1 },
+          undefined,
+          undefined,
+          ErrorDestination.User
+        );
+      }
+
+      assert(this.nodeDebugSession !== null);
```

Comment:
if this asser fails we won't send neither a response nor the error response. Maybe add this check to the `if` block above?

Reply:
I changed the assert to a warning, it doesn't seem too broken if its not set correctly. I think the only thing that will break in that case is the graceful shutdown of the proxy session if the JS session terminates unexpectedly.